sync_test.go

```diff
@@ -399,6 +399,23 @@ func testRescan(harness *neutrinoHarness, t *testing.T) {
 	}
 }
 
+func waitTx(t *testing.T, node *rpcclient.Client, hash chainhash.Hash) {
```

Comment:
cool, this lgtm 👍 

could also move this into an embedded function in the test itself since it is only used there. And we could even copy the pattern of `wait.Predicate(...)` from the lnd repo and write it like this (up to you but its nice that it waits a bit before calling the func the first time. But i like that your way uses `time.NewTimer` instead of `time.After` 👍 ):
```
waitTx := func(node *rpcclient.Client, hash chainhash.Hash) error {
		exitTimer := time.After(5 * time.Second)
		for {
			<-time.After(200 * time.Millisecond)

			select {
			case <-exitTimer:
				return fmt.Errorf("tx not found after time out")
			default:
			}

			if _, err := node.GetRawTransaction(&hash); err != nil {
				continue
			}

			return nil
		}
	}
```

Reply:
Made it a closure, and incorporated the initial delay as you indicated was nice, but left it as a test helper.